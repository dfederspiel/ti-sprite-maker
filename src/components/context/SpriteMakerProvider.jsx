import React, { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';
import { defaultGrid, getHex } from './constants';

export const useSpriteMakerProviderProps = () => {
  const [sprite] = useState('0'.repeat(16));

  return {
    sprite,
    getHex,
    defaultGrid,
  };
};

const spriteMakerContext = createContext(null);

export const SpriteMakerProvider = ({ children }) => {
  const props = useSpriteMakerProviderProps();
  return (
    <spriteMakerContext.Provider
      value={props}
    >
      {children}
    </spriteMakerContext.Provider>
  );
};

SpriteMakerProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};

export const useSpriteMaker = () => useContext(spriteMakerContext);
