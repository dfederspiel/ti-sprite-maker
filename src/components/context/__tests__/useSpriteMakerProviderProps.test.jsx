import { renderHook } from '@testing-library/react-hooks';
import { useSpriteMakerProviderProps } from '../SpriteMakerProvider';

describe('the sprite maker provider props', () => {
  describe('given it is initialized for the first time', () => {
    it('will initialize with a default sprite', () => {
      const { result } = renderHook(() => useSpriteMakerProviderProps());
      expect(result.current.sprite).toEqual('0'.repeat(16));
    });
  });
});
