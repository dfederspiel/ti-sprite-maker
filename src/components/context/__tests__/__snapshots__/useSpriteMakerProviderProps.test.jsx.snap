// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the sprite maker provider props exists 1`] = `
Object {
  "defaultGrid": Array [
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
  ],
  "getHex": [Function],
  "sprite": "0000000000000000",
}
`;
